data_safe_haven/allowlist/allowlist.py

```diff
@@ -0,0 +1,162 @@
+from difflib import unified_diff
+from typing import TypeVar
+
+from data_safe_haven.config import Context, DSHPulumiConfig, SREConfig
+from data_safe_haven.external import AzureSdk
+from data_safe_haven.infrastructure import SREProjectManager
+from data_safe_haven.types import AllowlistRepository
+
+T = TypeVar("T", bound="Allowlist")
+
+
+class Allowlist:
+    """Allowlist for packages"""
+
+    @classmethod
+    def from_remote(
+        cls: type[T],
```

Comment:
If you add `from typing import Self` above, you should be able to use `cls: Self` here instead of the TypeVar work-around. Added in Python 3.11.